src/main/kotlin/io/snyk/plugin/services/download/CliDownloaderErrorHandler.kt

```diff
@@ -14,16 +16,30 @@ class CliDownloaderErrorHandler {
         SnykBalloonNotificationHelper.showError(message, project,
             NotificationAction.createSimple("Retry CLI download") {
                 project.getService(SnykCliDownloaderService::class.java).downloadLatestRelease(indicator, project)
+                runBackgroundableTask("Retry Snyk CLI Download", project, false) {
+                    project.getService(SnykCliDownloaderService::class.java).downloadLatestRelease(it, project)
+                }
             },
             NotificationAction.createSimple("Contact support...") {
                 BrowserUtil.browse("https://snyk.io/contact-us/?utm_source=JETBRAINS_IDE")
             })
     }
 
     fun handleIOException(exception: IOException, indicator: ProgressIndicator, project: Project) {
-        val downloader = project.getService(SnykCliDownloaderService::class.java).downloader
-        downloader.downloadFile(getCliFile(), indicator)
-        showErrorWithRetryAndContactAction(getNetworkErrorNotificationMessage(exception), indicator, project)
+        retryDownload(project, indicator, getNetworkErrorNotificationMessage(exception))
+    }
+
+    private fun retryDownload(project: Project, indicator: ProgressIndicator, message: String) {
+        try {
+            runBackgroundableTask("Retry CLI Download", project, false) {
+                val downloader = project.getService(SnykCliDownloaderService::class.java).downloader
+                downloader.downloadFile(getCliFile(), indicator)
+            }
+        } catch (throwable: Throwable) { // we must catch throwable as IntelliJ could throw AssertionError
```

Comment:
I would agree here with detekt and check if we can catch more specific type of exceptions here

Reply:
Which class would you recommend? Unfortunately, AssertionError is in the error hierarchy, not in the exception hierarchy. Or would you prefer having two catch blocks?